### PR TITLE
Support AWS Local Zones

### LINKED ISSUE #4734: Cluster Creation in a Region with Local Zones
Story comment below https://github.com/weaveworks/eksctl/issues/4734#issuecomment-1068961392

---

**What feature/behavior/change do you want?**
When you try to create a cluster in an AWS Region that has one or more local zones enabled, and you are also creating the VPC as part of the cluster creation process, then eksctl passes the subnets in a local zone to the cluster. Unlike regional subnets, EKS can't place network interfaces into Local Zone subnets. As a result, the cluster creation process fails.

**Why do you want this feature?**
AWS customers are trying to use local zones in EKS clusters and are trying to create EKS clusters that include worker nodes in local zones. Currently, they have to create the VPC with regional and local zone subnets as a separate step and then explicitly specify only regional subnets in their configuration file while creating an EKS cluster. They would like to do this with a one step process.

### PR DESCRIPTION
**Pull request by cPu1**
### Description

Adds support for AWS Local Zones based on [this proposal](https://github.com/weaveworks/eksctl/issues/4734#issuecomment-1103809860). 

Closes #4734 

### Checklist
- [x] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes
- [x] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

**Review by Himangini — APPROVED** [review]
LGTM 👍🏻

File: pkg/apis/eksctl.io/v1alpha5/validation.go [hunk 1]
```diff
@@ -718,6 +733,10 @@ func ValidateNodeGroup(i int, ng *NodeGroup) error {
 		}
 	}
 
+	if len(ng.LocalZones) > 0 && len(ng.AvailabilityZones) > 0 {
+		return errors.New("cannot specify both localZones and availabilityZones")
```
Comment (Skarlso) [inline pkg/apis/eksctl.io/v1alpha5/validation.go:737]: How come? Can't you have a hybrid cluster with specific workloads running in a localzone vs some other nodegroup which defines availability zones that will take core a different workload?

Comment (Skarlso) [reply pkg/apis/eksctl.io/v1alpha5/validation.go:737]: Oh wait, never mind. This is the same nodegroup.

Comment (cPu1) [reply pkg/apis/eksctl.io/v1alpha5/validation.go:737]: >Can't you have a hybrid cluster with specific workloads running in a localzone vs some other nodegroup which defines availability zones that will take core a different workload?

You can. As you noticed, this validation is to disallow nodegroups having both local zones and AZs defined.

**Review by Skarlso — APPROVED** [review]
Well done! Nice work!

**Pull request merged**